### DISCUSSION TITLE
普通账号如何以知名端口（80等）启动服务

### DISCUSSION THREAD

**Discussion by wang3develop** [linuxdeepin/developer-center#3294] [Q&A | 问答板块] [answered]
之前发过一个帖子https://bbs.deepin.org/zh/post/239954。
现在我专门弄了一个测试的程序。
[test.zip](https://github.com/linuxdeepin/developer-center/files/9052232/test.zip)

希望对于普通账号，使用80、443等知名端口号 有合适的解决方法！

**Comment by BLumia** [comment, marked as answer]
非 root 权限可以用 [`CAP_NET_BIND_SERVICE`](http://man7.org/linux/man-pages/man7/capabilities.7.html) 获得 <1024 的端口绑定，也可以参考[这个答案](https://superuser.com/questions/710253/allow-non-root-process-to-bind-to-port-80-and-443/892391#892391)。

p.s. 作为补充，在运行服务时，推荐的做法是遵循最小权限原则，给予其能正常工作的最小权限。方式包括另开账号并配置最小的权限，使用对应的账号运行服务之类。是要规避直接使用 root **账户**允许，并不是说完全不能给予 root **权限**。另外，如果你跑服务的机器上有 http 服务器（比如 apache 或 nginx），建议由它管理 80/443 端口，然后把流量以反向代理的形式转发到你的程序中。
